packages/controls/src/controls/typography/typography.ts

```diff
@@ -0,0 +1,180 @@
+import { z } from 'zod'
+
+import { isNotNil } from '../../lib/functional'
+import { safeParse, type ParseResult } from '../../lib/zod'
+
+import { type CopyContext } from '../../context'
+import { Targets, type IntrospectionTarget } from '../../introspection'
+import {
+  type DeserializedRecord,
+  type SerializedRecord,
+} from '../../serialization'
+
+import { ControlDefinition, serialize } from '../definition'
+import { DefaultControlInstance, type SendMessage } from '../instance'
+
+import * as Schema from './schema'
+
+type Schema = typeof Definition.schema
+type DataType = z.infer<Schema['data']>
+type ValueType = z.infer<Schema['value']>
+type ResolvedValueType = z.infer<Schema['resolvedValue']>
+type InstanceType = DefaultControlInstance
+
+class Definition extends ControlDefinition<
+  typeof Definition.type,
+  unknown,
+  DataType,
+  ValueType,
+  ResolvedValueType,
+  InstanceType
+> {
+  static readonly type = 'makeswift::controls::typography' as const
+
+  static get schema() {
+    const type = z.literal(Definition.type)
+
+    const data = Schema.typography(
+      z
+        .object({
+          swatchId: z.string().nullable(),
+          alpha: z.number().nullable(),
+        })
```

Comment:
@migueloller Here's the nullable color data definition that you asked about earlier.